upload-tsunami-package

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+################################################################################
+#
+# Helper script to upload tsunami libmxnet packages to the correct bintray repo
+#
+################################################################################
+#
+# It is assumed that the default `UPLOAD_DESTINATION` for package releases is
+# already specified in Travis CI config; this script overrides it for package
+# versions containing `alpha`, `beta`, or `rc` (whether this is part of the
+# upstream version or the package version), and uploads either to the default
+# or prerelease repo accordingly.
+#
+# This script is licensed under the terms of the Boost Software License,
+# Version 1.0.
+#
+################################################################################
+
+set -ex
+
+if echo "$TRAVIS_TAG" | grep -qe "\<alpha\|beta\|rc\>"; then
```

Comment:
Locally (in my shell) I need to remove the escaping of `|` to make `grep` work. But this might be only a local thing.

Reply:
man grep:
```
   Basic vs Extended Regular Expressions
       In  basic  regular  expressions  the meta-characters ?, +, {, |, (, and ) lose their special meaning; instead use the backslashed
       versions \?, \+, \{, \|, \(, and \).
```
Maybe you have some alias to make grep use extended expressions by default? In any case I think it would be better to use single quotes (`'`), as it is a bit unclear if the shell should convert `"\|"` to `|` or leave it as `\|`.

Reply:
@leandro-lucarella-sociomantic so have the line read:

```sh
if echo "$TRAVIS_TAG" | grep -qe '\<alpha\|beta\|rc\>'; then
```
... ?

Reply:
Yup

Reply:
I forgot that I aliased `grep` to `egrep` which explains the local differences in behavior.